rclpy/CMakeLists.txt

```diff
@@ -14,8 +14,10 @@ find_package(PythonExtra MODULE)
 set(_PYTHON_EXECUTABLE "${PYTHON_EXECUTABLE}")
 if(NOT WIN32)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
-elseif(CMAKE_BUILD_TYPE STREQUAL "Debug")
-  set(PYTHON_EXECUTABLE "${PYTHON_EXECUTABLE_DEBUG}")
+else(NOT WIN32)
+  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(PYTHON_EXECUTABLE "${PYTHON_EXECUTABLE_DEBUG}")
+  endif()
```

Comment:
This patch is not changing the logic of the CMake code as far as I understand it. The Python executable variable gets updated when being on Windows (`else NOT WIN32`) and using debbug build type (`CMAKE_BUILD_TYPE STREQUAL "Debug"`) - before and after the patch. Can you please elaborate why this change should make a difference?